OpenOversight/app/main/views.py

```diff
@@ -74,26 +76,37 @@ def submit_complaint():
                            officer_image=request.args.get('officer_image'))
 
 
-@main.route('/submit')
+@main.route('/submit', methods=['GET', 'POST'])
+@login_required
 def submit_data():
-    return render_template('submit.html')
-
-
-@main.route('/upload', methods=['POST'])
-def upload_file():
-    file = request.files['file']
-    if file and allowed_file(file.filename):
-        filename = secure_filename(file.filename)
-        file.save(os.path.join(current_app.config['UNLABELLED_UPLOADS'], filename))
-        return redirect(url_for('main.show_upload',
-                                filename=filename))
-
-
-@main.route('/show_upload/<filename>')
-def show_upload(filename):
-    # return send_from_directory('../'+config.UNLABELLED_UPLOADS,
-    #                           filename)
-    return 'Successfully uploaded: {}'.format(filename)
+    form = HumintContribution()
+    if request.method == 'POST' and form.validate_on_submit():
+        filename = secure_filename(request.files[form.photo.name].filename)
+        image_data = request.files[form.photo.name].read()
+
+        # See if there is a matching photo already in the db
+        hash_img = hash_file(image_data)
+        hash_found = Image.query.filter_by(hash_img=hash_img).first()
+
+        if not hash_found:
```

Comment:
say someone uploads two files named 'Photo01.jpg' with different hashes simultaneously. We run in gunicorn so each thread runs without shared state excepting the filesystem.
`filename = secure_filename('Photo01.jpg')`, then we hash the image(s), then we check the database for hash duplicates, and then we write it out to the filesystem at '/tmp/Photo01.jpg'. Which one gets uploaded?

Also, we're trusting workzeug's `safe_filename`, which is about things that are safe to drop into a folder but not necessarily safe to drop into /tmp itself. What if someone manages to clobber something important-ish (like, say, a socket) in /tmp this way? Can we use the `tempfile` package in stdlib to specify the local filename to use instead?